scipy/spatial/distance.py

```diff
@@ -2190,6 +2191,20 @@ def pdist(X, metric='euclidean', *, out=None, **kwargs):
 
         if metric_info is not None:
             pdist_fn = metric_info.pdist_func
+            # TODO: remove out/w checks when distutils
+            # build system is removed
+            if os.name == "nt" and out is not None:
+                if out.size != (m * (m - 1) / 2):
```

Comment:
I don't think this checks number of dimensions
```suggestion
                if out.shape != (m * (m - 1) / 2,):
```